### ISSUE TITLE
Sound doesn't start right away: Chrome/Touch

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/ohms-law#139]
**Test device:**
Dell
**Operating System:**
Win 10
**Browser:**
chrome
**Problem description:**
For https://github.com/phetsims/QA/issues/313
When first moving the sliders with a touch screen on a Win 10 device with chrome, the sound will not play, or only play the end.  This does not seem to happen on other sims or on any of the published sims I looked at.
**Steps to reproduce:**
1. Do a fresh reload of the sim
2. Use the touch screen to move a slider

**Screenshots:**
https://drive.google.com/file/d/1P5NSvSi88ZlqiSNgSsARt_iAZGSafYcq/view?usp=sharing

Troubleshooting information (do not edit):

<details>

Name: ‪Ohm's Law‬
URL: https://phet-dev.colorado.edu/html/ohms-law/1.4.0-rc.1/phet/ohms-law_en_phet.html
Version: 1.4.0-rc.1 2019-05-01 19:41:33 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.131 Safari/537.36
Language: en-US
Window: 1536x722
Pixel Ratio: 2.5/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by jessegreenberg** [comment]
I am able to reproduce on the RC but I am *not* able to reproduce on master.

**Comment by jessegreenberg** [comment]
I am actually seeing this in the published version of RIAW too. Handing over to @jbphet since this is sound related.

**Comment by jbphet** [comment]
I've done some investigation on this, and the reason that sound doesn't start in this situation is that the `touchstart` event in `BrowserEvents.js` is not set up to trigger `scenery.Display.userGestureEmitter`.  The `touchend` event **is** set up for this.  I asked @jonathanolson about this, and the following Slack dialog ensued (this is edited somewhat for clarity):

```
@jbphet [11:57 AM]
Okay, here's the question: There is an issue logged that sound is not starting right away when the first user interactions are via a touch screen on a Win device in Chrome.  I can duplicate, and what I've found is that Display.userGestureEmitter doesn't fire on the touch start event, but DOES fire at touch end, which is when sound is getting turned on.  Is that an easy thing to change about Display.userGestureEmitter?

@jonathanolson [12:00 PM]
I’ll check, touchstart may not be on the list of events
I recall a Google doc linked from the userGestureEmitter issue that has a comprehensive list
I’ll see if I can pull it up

@jbphet [12:04 PM]
I just looked in BrowserEvents.js, and the touchstart does not call `scenery.Display.userGestureEmitter.emit()`, but touchend does.  Should I just add it, or might there be undesirable side effects?

@jonathanolson [12:05 PM]
Do not add it without checking. Could be undesirable
I’m looking now
Image from iOS 

So touchstart is very much not on the list
https://docs.google.com/document/d/1mcxB5J_u370juJhSsmK0XQONG2CIE3mvu827O-Knw_Y/edit#heading=h.t0ybg8byy7bg
At least for Chrome

@jbphet [12:13 PM]
Okay, so does that mean the emitter can't emit on that event?  Is there something else that I can and should do to pick up that event?

@jonathanolson [12:14 PM]
It means you can’t start audio on a touchstart I believe
So we wouldn’t want the emitter to fire then

@jbphet [12:15 PM]
Wait, I don't understand.  I just stuck the emit call in there as a test, and it starts sound just fine.  So it is just some sort of rule?  Because it seems to work.

@jonathanolson [12:34 PM]
In a situation where the security constraint is there (so if you try to init sound on load it would fail)

@jbphet [2:15 PM]
Sorry I disappeared there - I needed to step away for a while.  So my apologies, but this still isn't clear to me.  I don't quite understand your last comment.  Right now, in Chrome, the code can never initialize sound on load - that's why the `soundManager` is set up to wait for some sort of user interaction before enabling sound by "resuming" the audio context.  In Chrome on a Windows machine, `touchstart` currently doesn't trigger `userGestureEmitter` to emit because there is no line of code to cause it to emit.  I tried adding that line of code, and on my machine, it solves the problem.
So are you recommending **not** adding that line of code and, if so, why?
```

Assigning to @jonathanolson to answer the question posed at the end of the dialog.

**Comment by jbphet** [comment]
I understand now what @jonathanolson was saying.  The gist is this: Chrome has a list of user events that are essentially "approved" as events that can be used to resume the audio context.  `touchstart` and `pointerdown` - when it comes from a touch event - are not on this list.  Perhaps the thinking is that users accidentally touch their screens all the time, and they don't want that starting up audio.  At any rate, I locally modified `BrowserEvents.js` so that it would fire `Display.userGestureEmitter` on these events and, sure enough, Chrome would not let the audio context be resumed.

So, the bottom line is that we can't fix this, because it's a fundamental limitation of the browser.

I tried testing this on Firefox and Edge on my machine, but both of them allow the audio context to be running at startup.  I'm not sure why, but a little bit of web searching led me to believe that maybe they are being more selective about what they disable, and it's more specific to auto-loading videos.  This is an educated guess on my part, and all I know for sure is that sound isn't disabled on load time for me on these two browsers.

I don't currently have a device available to test touch on a Mac running Safari, but it doesn't matter much, because if it's the same as Chrome, there isn't anything we can do anyway.

I'm marking this as "won't fix" and assigning back to @KatieWoe and @jessegreenberg so that they can be aware of this situation.  If neither have any questions, this can be closed.

For the record, I did make some commits related to this that were designed to improve the logging that I use to debug audio-context-related issues.  These messages can be seen by using the "log" query parameter.
